website/pages/docs/nia/architecture.mdx

```diff
@@ -6,4 +6,38 @@ description: >-
   Consul-Terraform-Sync Architecture
 ---
 
-# Consul-Terraform-Sync Architecture
+# Consul Terraform Sync Architecture
+
+Consul Terraform Sync is a service-oriented tool for managing
+network infrastructure near real-time. Consul-Terraform-Sync runs as a daemon
+and integrates the network topology maintained by your Consul cluster with your
+network infrastructure to dynamically secure and connect services.
+
+## 10,000 Foot View
+
+[![Consul Terraform Sync Architecture](/img/nia-highlevel-diagram.png)](/img/nia-highlevel-diagram.png)
+
+The diagram shows Consul Terraform Sync monitoring the Consul service catalog
+for updates and utilizing Terraform to update the state of the infrastructure.
+
+There are two principal aspects of Sync to know about corresponding to the
+lines to Consul and Terraform in the diagram above. The line to Consul
+represents the Watches monitoring the state of Consul's service catalog (and
+possibly KV store)while the line to Terraform represents Tasks being run to
+update the infrastructure.
+
+## Watcher and Views
+
+Consul-Terraform-Sync monitors Consul for updates utilizing Consul's [Blocking
+Queries](/api-docs/features/blocking) whenever supported, falling back on
+polling when not. The watcher maintains a separate thread (known internally as
+a view) for each value monitored, running any Tasks that depend on that watched
+value whenever it's updated.
+
+## Tasks
+
+A task is the translation of dynamic service information from the Consul
```

Comment:
This section reads too close to the Tasks page, is there a different angle you could take for including it under Architecture? https://nia-docs-tech-preview--consul-docs-preview.netlify.app/docs/nia/tasks

Reply:
Similar as we had the same source. I can try to reword it again.

Reply:
lol. let me know if I should do rewording in the task section. to have different content, do you think it would be worth it to document here (or in the task section) how tasks run asynchronously and all that fun stuff?

Reply:
Not sure if it's any good... but it is different now.